moveit_setup_assistant/moveit_setup_srdf_plugins/src/robot_poses.cpp

```diff
@@ -0,0 +1,157 @@
+  request.max_contacts = 1;
+  request.max_contacts_per_pair = 1;
+  request.verbose = false;
+}
+
+// ******************************************************************************************
+// Find the associated data by name
+// ******************************************************************************************
+srdf::Model::GroupState* RobotPoses::findPoseByName(const std::string& name, const std::string& group)
+{
+  // Find the group state we are editing based on the pose name
+  srdf::Model::GroupState* searched_state = nullptr;  // used for holding our search results
+
+  for (srdf::Model::GroupState& state : srdf_config_->getGroupStates())
+  {
+    if (state.name_ == name && state.group_ == group)  // match
+    {
+      searched_state = &state;
+      break;
+    }
+  }
+
+  return searched_state;
+}
+
+// ******************************************************************************************
+// Load the allowed collision matrix from the SRDF's list of link pairs
+// ******************************************************************************************
+void RobotPoses::loadAllowedCollisionMatrix()
+{
+  // Clear the allowed collision matrix
+  allowed_collision_matrix_.clear();
+
+  // Update the allowed collision matrix, in case there has been a change
+  for (const auto& disabled_collision : srdf_config_->getDisabledCollisions())
+  {
+    allowed_collision_matrix_.setEntry(disabled_collision.link1_, disabled_collision.link2_, true);
+  }
+}
+
+// ******************************************************************************************
+// Publish the current RobotState to Rviz
+// ******************************************************************************************
+bool RobotPoses::publishState(const moveit::core::RobotState& robot_state)
+{
+  // Create a planning scene message
+  moveit_msgs::msg::DisplayRobotState msg;
+  moveit::core::robotStateToRobotStateMsg(robot_state, msg.state);
+
+  // Publish!
+  pub_robot_state_->publish(msg);
+
+  // Decide if current state is in collision
+  collision_detection::CollisionResult result;
+  srdf_config_->getPlanningScene()->checkSelfCollision(request, result, robot_state, allowed_collision_matrix_);
+  return !result.contacts.empty();
+}
+
+std::vector<const moveit::core::JointModel*> RobotPoses::getSimpleJointModels(const std::string& group_name) const
+{
+  moveit::core::RobotModelPtr robot_model = srdf_config_->getRobotModel();
+  if (!robot_model->hasJointModelGroup(group_name))
+  {
+    throw std::runtime_error(std::string("Unable to find joint model group for group: ") + group_name +
+                             ". Are you sure this group has associated joints/links?");
+  }
+
+  const moveit::core::JointModelGroup* joint_model_group = robot_model->getJointModelGroup(group_name);
+
+  std::vector<const moveit::core::JointModel*> joint_models;
+  for (const moveit::core::JointModel* joint_model : joint_model_group->getJointModels())
```

Comment:
This loop basically filters for [active joint models](https://github.com/ros-planning/moveit2/blob/main/moveit_core/robot_model/include/moveit/robot_model/joint_model_group.h#L156), except maybe for the 1-variable joints. But it's probably not necessary to change anything over this.